src/transfer.cpp

```diff
@@ -777,6 +785,12 @@ void Transfer::complete()
         LOG_debug << "Upload complete: " << (files.size() ? files.front()->name : "NO_FILES") << " " << files.size();
 
         // files must not change during a PUT transfer
+        if (slot->fa->asyncavailable())
+        {
+            slot->fa->closef();
+            slot->fa->fopen(&localfilename);
```

Comment:
Yes, it's needed to open the file to be used synchronously there. Anyway, I have removed the `closef()` because that function should be for internal use only. Instead, I remove the object and create a new one.